src/main/java/org/sonar/scm/git/blame/BlobReader.java

```diff
@@ -79,7 +80,7 @@ private RawText loadText(String path) throws IOException {
     // FileTreeIterator's InputStream, which filters certain characters. For example, it removes windows lines terminators
     // in files checked out on Windows.
     TreeWalk treeWalk = new TreeWalk(repository);
-    treeWalk.addTree(new FileTreeIterator(repository));
+    addTreeToTreeWalk(treeWalk, repository);
```

Comment:
I tested your version in SLCore on a bare repository, and it failed. In a bare repo, there is no working directory, meaning no working files. To see the diff, all commits need to be calculated from the beginning. In this version, there is a FileIterator (line 87), which fails for bare repos, since there is no file to iterate.